sdk/servicebus/azure-servicebus/tests/perf_tests/_test_base.py

```diff
@@ -5,124 +5,208 @@
                     batch = await sender.create_message_batch()
                     batch.add_message(ServiceBusMessage(data))
-            await sender.send_messages(batch)
 
-    async def global_setup(self):
+            if len(batch):
+                await sender.send_messages(batch)
+    
+    async def global_setup(self) -> None:
         await super().global_setup()
-        await self._preload_queue()
+        await self._preload_topic()
     
-    async def close(self):
+    async def close(self) -> None:
         self.receiver.close()
         await self.async_receiver.close()
         await super().close()
+    
 
+class _SendTest(BatchPerfTest):
+    def __init__(self, arguments) -> None:
+        super().__init__(arguments)
+        transport_type=TransportType.AmqpOverWebsocket if arguments.transport_type == 1 else TransportType.Amqp
+
+        self.connection_string=self.get_from_env("AZURE_SERVICEBUS_CONNECTION_STRING")
+        self.service_client=ServiceBusClient.from_connection_string(
+            self.connection_string,
+            transport_type=transport_type,
+            uamqp_transport=arguments.transport_type,
+        )
+        self.async_service_client=AsyncServiceBusClient.from_connection_string(
+            self.connection_string,
+            transport_type=transport_type,
+            uamqp_transport=arguments.transport_type,
+        )
+    async def close(self) -> None:
+        self.service_client.close()
+        await self.async_service_client.close()
+        await super().close()
+    
     @staticmethod
-    def add_arguments(parser):
-        super(_ReceiveTest, _ReceiveTest).add_arguments(parser)
-        parser.add_argument('--peeklock', action='store_true', help='Receive using PeekLock mode and message settlement.', default=False)
-        parser.add_argument('--max-wait-time', nargs='?', type=int, help='Max time to wait for messages before closing. Defaults to 0.', default=0)
-        parser.add_argument('--preload', nargs='?', type=int, help='Number of messages to preload. Default is 10000.', default=10000)
+    def add_arguments(parser) -> None:
+        super(_SendTest, _SendTest).add_arguments(parser)
+        parser.add_argument('--message-size', nargs='?', type=int, help='Size of a single message. Defaults to 100 bytes', default=100)
+        parser.add_argument('--batch-size', nargs='?', type=int, help='The number of messages that should be included in each batch. Defaults to 100', default=100)
+        parser.add_argument('--uamqp-transport', action="store_true", help="Switch to use uamqp transport. Default is False (pyamqp).", default=False)
+        parser.add_argument('--transport-type', nargs='?', type=int, help="Use Amqp (0) or Websocket (1) transport type. Default is Amqp.", default=0)
+        parser.add_argument('--no-client-share', action='store_true', help='Create one ServiceClient per test instance.  Default is to share a single ServiceClient.', default=False)
```

Comment:
Not sure we need this one?

Reply:
No, we can remove this since every run is getting their own client.